### PR TITLE
(#5635) Add TLS configuration in the ingress trait

### PR DESCRIPTION
**Pull request by romain-pfund**
Add 
- tlsHosts
- tlsSecretName 
to configure TLS in the ingress trait

**Release Note**
```release-note
Ingress Trait: Add tlsHosts and tlsSecretName to configure the TLS ingress spec
```

### CODE REVIEW

File: pkg/trait/ingress.go [hunk 1]
```diff
@@ -126,6 +128,15 @@ func (t *ingressTrait) Apply(e *Environment) error {
...
 	}
 
+	if len(t.TLSHosts) > 0 || len(strings.TrimSpace(t.TLSSecretName)) > 0 {
+		ingress.Spec.TLS = []networkingv1.IngressTLS{
+			{
+				Hosts:      t.TLSHosts,
+				SecretName: t.TLSSecretName,
+			},
+		}
+	}
```
Comment (romain-pfund) [inline pkg/trait/ingress.go:138]: @squakez :
like I said, not a Go developper.
They may be a better way to do this ?

What we want, is to completely not include the ingress.Spec.TLS if TLSHosts or TLSSecretName is not set

**Comment by romain-pfund** [comment]
I haven't had the time to fully check with a working certificat

but the kubect get ingress xxx -o yaml seams fine when comparing to an internal working ingress (not showing here for confidentiality reason):

image

image
image

**Comment by romain-pfund** [comment]
> LGTM, thanks for the work. We need also to add the outcome of `make generate` though.
ok, so I run
```
make
make generate
```
and commit/push ?

**Comment by squakez** [comment]
> I haven't had the time to fully check with a working certificat
> 
> but the kubect get ingress xxx -o yaml seams fine when comparing to an internal working ingress (not showing here for confidentiality reason):

Sure, the important work here is to map into the configuration expected by Kubernetes.

**Comment by squakez** [comment]
> ```
> make
> make generate
> ```
> 
> and commit/push ?

Yes, you should see several autogenerated resources. They must be added to the PR, you can `git commit --amend` and force push this. Also, you can see there is some minor lint error due to codestyle formatting. The check is failing with:

```
pkg/trait/ingress.go:22  goimports  File is not `goimports`-ed
```

You can perform a `gofmt -w pkg/trait/ingress.go` and later a `make lint` to verify that is solved afterward. And again, commit the change (likely some space format).

File: .github/actions/gh-go-mod-generate-sbom
Comment (squakez) [inline .github/actions/gh-go-mod-generate-sbom:1]: Please, remove this one from the commit.

File: pkg/trait/ingress.go (hunk 1 above)
Comment (romain-pfund) [reply pkg/trait/ingress.go:138]: ok, doubled check with the doc, and yes, need both.
so I'll change to "and" and add more tests

File: .github/actions/gh-go-mod-generate-sbom
Comment (romain-pfund) [reply .github/actions/gh-go-mod-generate-sbom:1]: removed

File: pkg/trait/ingress.go (hunk 1 above)
Comment (romain-pfund) [reply pkg/trait/ingress.go]: replace by 
len(t.TLSHosts) > 0 && t.TLSSecretName != ""
+ 2 test added to cover both part of the if

**Comment by romain-pfund** [comment]
Some builds are failing
Did i missed something ?

**Comment by squakez** [comment]
> Some builds are failing Did i missed something ?

Install check is failing because of https://github.com/apache/camel-k/issues/5632 (I'm working on a fix). The rest should be working normally. I gave it another shot.

**Review by squakez — APPROVED** [review]

**Comment by squakez** [comment]
Thanks for the contribution!

**Pull request merged**